docs/proposals/odo-config-file.md

```diff
@@ -0,0 +1,223 @@
+
+## Design overview
+
+We need to provide a configuration file, `.odo`, that will hold specific configuration that `odo` will use on execution. This configuration might modify how the CLI work or how the component will work on the platform.
+
+### Scopes of configuration
+
+`odo` CLI configuration needs to be preserved in configuration files. As a developer might work on multiple
+components, there is a need to have different scopes for configuration so that a developer don't need to
+explicitly configure each and every component independently.
+
+#### Local configuration
+
+Every component can be configured independently. We call this local configuration for the component as it only affects the component it references. This configuration will be stored in a file in the root directory of the component's source code. In this way, the configuration file can be added to version control system and preserved along with the application. If any other developer checks out the same source code he will get the same configuration as the original developer.
+
+A good representation of this is a developer that while on the development process realizes his application requires 2 GB of memory to run appropriately, he might decide to save that information in this configuration file. At a later moment, any other developer (or even him in the future) that decides to work on the same application will have this configuration definition along with his source code. `odo` should take into account this configuration definition and materialize when appropriate. The type of configuration, format and applicability will be described later in this document.
+
+#### Global configuration
+
+When working with multiple components it might be tedious to configure each and every component, or a user wants to have a default configuration value applied to each and every component he works with when no explicit (local) configuration exists. For this purpose, there should be a file, stored in the user's home directory, that should contain this configuration definitions. This configuration is global to every component the user works with.
+
+#### User explicit configuration
+
+Often times a user wants to use different configuration values than the provided in the configuration file. We need to provide a way to allow a developer to specify this new values he want to use. For that, the user needs to explicitly provide he values as CLI arguments to the speficic `odo` command.
+
+An example of this would be:
+
+```bash
+odo create wildfly --memory 2GB
+```
+
+In this previous example, the user is explicitly instructing `odo` to use 2GB of memory when creating a component based on wildfly.
+
+### <a name="configuration-precedence"></a> Configuration precedence
+
+As configuration can exist both localy and globaly or only on one of these scopes, it seems obvious that there needs to be some rule managing the precedence and applicability of the configuration.
```

Comment:
just a minor spell check,
`locally` and `globally`

Reply:
Fixed.